### PR TITLE
feat/cli

### PR DESCRIPTION
**Pull request by weaintplastic**
- [x] Bootstrap package.json
- [x] Let template folder overwrite existing files
- [x] Delete files/directories that should be cleared
- [x] Add more validation for if it is secure to create project in the desired location
- [x] Transform given keywords into array
- [x] Validate author
- [x] Validate package name
- [x] Validate version name
- [x] Bootstrap package name with given directory name

### CODE REVIEW

**Comment by weaintplastic** [comment]
You can test as following

```
$ npm install -g @researchgate/node-package-blueprint@0.3.0-4

$ node-package-blueprint <some-directory>
```

File: src/js/bin/index.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    }
+    return false;
+};
+
+const createProject = (name, config) => {
+    const targetDir = path.resolve(name);
+    const sourceDir = path.resolve(__dirname, '../../../');
+    const appName = path.basename(targetDir);
+
+    console.log();
```
Comment (Rendez) [inline src/js/bin/index.js]: Maybe replace all `console.log()` with `log()`, adding at the top of the file `const log = console.log`

File: src/js/bin/index.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        .then(() => {
+            return fs.copy(path.resolve(__dirname, 'template'), targetDir);
+        })
+        .then(() => {
+            console.log();
+            console.log();
+            console.log(`🙌🏻  Your project is ready and is waiting for you over here: ${targetDir}`);
+            console.log();
+            spinner.stop();
+        });
```
Comment (Rendez) [inline src/js/bin/index.js]: Ideally we catch here anything going wrong

File: src/js/bin/index.js [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+            console.log('📦  Prepare project');
+            return fs.copy(`./tmp`, targetDir, {
+                filter: (src, dest) => {
+                    const relativePath = path.relative(`./tmp`, src);
+                    return !shouldBeIgnored(relativePath);
+                },
+            });
+        })
+        .then(() => {
+            return fs.outputFile(path.join(targetDir, 'package.json'), JSON.stringify(newPackageJson, null, 2));
```
Comment (Rendez) [inline src/js/bin/index.js:195]: `outputFileSync` and you don't need the timeout

File: src/js/bin/index.js [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+const timeout = () =>
+    new Promise(resolve => {
+        setTimeout(resolve, 1000);
+    });
+
+const isNameValid = name => {
+    const validationResult = validateProjectName(name);
+    if (!validationResult.validForNewPackages) {
+        console.error(
```
Comment (Rendez) [inline src/js/bin/index.js:221]: `throw new Error()` works along with the `.catch()` to create a single error handling point.

**Pull request merged**